flash/data/data_pipeline.py

```diff
@@ -0,0 +1,224 @@
+        """
+
+        super_obj = DataPipeline
+
+        if not hasattr(self, method_name) or not hasattr(super_obj, method_name):
+            return False
+
+        return getattr(self, method_name).__code__ is not getattr(super_obj, method_name)
+
+    @staticmethod
+    def _do_nothing_collate(samples: Sequence[Any]) -> Sequence[Any]:
+        return samples
+
+    def split_around_collate(self, collate_fn: Optional[Callable] = None) -> Tuple[Collater, Collater]:
+
+        if collate_fn is None:
+            collate_fn = default_collate
+
+        post_collate_overriden = self._is_overriden('post_collate')
+        device_pre_collate_overriden = self._is_overriden('device_pre_collate')
+
+        if post_collate_overriden and device_pre_collate_overriden:
+            raise MisconfigurationException(
+                f'{self.__class__.__name__}: post_collate and gpu_pre_collate are mutual exclusive.'
+            )
+
+        elif post_collate_overriden:
+            worker_collate = collate_fn
+            device_collate = self._do_nothing_collate
+
+        elif device_pre_collate_overriden:
+            worker_collate = self._do_nothing_collate
+            device_collate = collate_fn
+
+        else:
+            worker_collate = collate_fn
+            device_collate = self._do_nothing_collate
+
+        worker_callable = Collater(worker_collate, self.pre_collate, self.post_collate)
+        device_callable = Collater(device_collate, self.device_pre_collate, self.device_post_collate)
+
+        return worker_callable, device_callable
+
+    @staticmethod
+    def _model_transfer_to_device_wrapper(func: Callable, collater: Collater) -> Callable:
+
+        @wraps(func)
+        def new_func(*args, **kwargs):
+            moved_to_device = func(*args, **kwargs)
+            return collater(moved_to_device)
+
+        return new_func
+
+    def _attach_to_model(self, model: LightningModule, loader_stage: str = 'all') -> LightningModule:
+        if loader_stage == 'all':
+            loader_stage = ['train', 'test', 'val', 'predict']
+
+        elif isinstance(loader_stage, str):
+            loader_stage = [loader_stage]
+
+        for stage in loader_stage:
+            loader_name = f'{stage}_loader'
+
+            if hasattr(model, loader_name):
+                dataloader = getattr(model, loader_name)
+
+                if isinstance(dataloader, _PatchDataLoader):
+                    wrap_patch_loader = True
+                    dataloader = dataloader()
+
+                else:
+                    wrap_patch_loader = False
+
+                if isinstance(dataloader, Sequence):
+                    was_seq = True
+                else:
+                    dataloader = [dataloader]
+                    was_seq = False
+
+                for idx, loader in enumerate(dataloader):
+                    if isinstance(loader, DataLoader):
```

Comment:
This won't work with custom dataloader. See data_loading.py in Lightning.

Reply:
That's why it's guarded like this. But IMO we shouldn't expect any custom loaders here, since then people would be using lightning. Also you cannot pitch this for custom loaders without knowing their internals

Reply:
I don't agree with that I just ran into a huge issue with this today if I want to extend flash capabilities I shouldn't have to implement a lightning datamodule from scratch to take advantage of lightnings features.

Reply:
Yes, but what kind of interface do you want to assume in that case? E.g. if you have a custom loader class, there might not even be something we can attach to...